gui/src/components/deviceManagement/ManageDeviceConfiguration.js

```diff
@@ -78,23 +95,100 @@ class ManageDeviceConfiguration extends React.Component {
     return configurationFields;
   }
 
-  toggleDeletionModal = () => {
-    console.log('hi')
-    this.setState({deleteIsOpen: !this.state.deleteIsOpen});
+
+  toggleDeletionConfirmationModal = () => {
+    this.setState({deleteConfirmIsOpen: !this.state.deleteConfirmIsOpen});
+  }
+
+  toggleModifyConfirmationModal = () => {
+    this.setState({saveConfirmIsOpen: !this.state.saveConfirmIsOpen});
   }
+  
+  modifyDevice = () => {
+    xhr.open('PUT', modifyDeviceConfigurationBaseUrl + "?networkName=" + this.state.originalNetworkName + "&deviceId=" + this.state.originalDeviceName);
+    xhr.setRequestHeader("Content-Type", "application/json");
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === XMLHttpRequest.DONE) { // Once the request is done
+        if (xhr.status === 200) {
+          this.setState({
+            saveConfirmIsOpen: false
+          });
+          this.setState({
+            saveSuccessIsOpen: !this.state.successIsOpen
+          });      
+        }
+        else {
+          console.log(xhr);
+          console.log(xhr.response);
+        }
+      }
+    }    
+
+
+    var data = {};
+    var temp = [this.state.newDeviceProperties['Tag Population']];
 
-  toggleSuccessModal = () => {
+    // ======= When creating fields, no reference to field type, so some fields are would be sent as strings when they need to be ints. Also antenna beeds array. Need to fix ========
+
+    if (this.state.originalDeviceName == this.state.newDeviceProperties["Device Name"]) {
+      data = {
+        "networkName": this.state.originalNetworkName,
+        "deviceId": this.state.newDeviceProperties['Device Name'],
+        //"newDeviceId": this.state.newDeviceProperties['IP Address'], // Not needed because not changing name
```

Comment:
could probably do better than repeating all the fields just to hide newDeviceId in if/else

Reply:
https://github.com/chariot-dev/chariot/commit/cb084c65ea95c0d1a38bd7d43b9ffeacc78c0a9a